tennis-scheduler-web-app/src/components/Login.jsx

```diff
@@ -1,24 +1,33 @@
 import React, { useState, useEffect } from "react";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { login } from "../api/PersonApi";
 
 export const Login = () => {
   const [formValues, setFormValues] = useState({ email: "", password: "" });
   const [formErrors, setFormErrors] = useState({});
-  const [isSubmit, setIsSubmit] = useState(false);
   const [loginMessage, setLoginMessage] = useState("");
+  const navigate = useNavigate();
 
-  const loginSubmit = (e) => {
+  const loginSubmit = e => {
     e.preventDefault();
     setFormErrors(validation(formValues));
-    //there will be login request, and loginMessage will be set if login is failed
+
+    if (Object.keys(formErrors).length === 0) {
+      login(formValues).then((response) => {
+        localStorage.setItem("token", response.data.accessToken);
+        localStorage.setItem("role", response.data.role);
```

Comment:
If there is role in token, you could extract it from there, if you save it into local storage its unsafe and user could change it

Reply:
We don't have role in token :(

Reply:
Then add it, its much safer than this

Reply:
I  will change that on backend develop branch, to avoid opening PR-s if that's ok?

Reply:
New branch for everything you change, even if that is small fix